src/main/scala/ai/privado/languageEngine/go/tagger/collection/CollectionTagger.scala

```diff
@@ -0,0 +1,150 @@
+package ai.privado.languageEngine.go.tagger.collection
+
+import ai.privado.cache.RuleCache
+import ai.privado.model.{Constants, InternalTag, RuleInfo}
+import ai.privado.tagger.PrivadoParallelCpgPass
+import io.shiftleft.codepropertygraph.generated.Cpg
+import io.shiftleft.codepropertygraph.generated.nodes.Method
+import io.shiftleft.semanticcpg.language.*
+import ai.privado.utility.Utilities.*
+
+import scala.collection.mutable
+import scala.util.{Failure, Success, Try}
+
+class CollectionTagger(cpg: Cpg, ruleCache: RuleCache) extends PrivadoParallelCpgPass[RuleInfo](cpg) {
+  private val methodUrlMap        = mutable.HashMap[Long, String]()
+  private val classUrlMap         = mutable.HashMap[Long, String]()
+  private val ROUTES_FILE_PATTERN = ".*(routes|routers).go"
+
+  override def generateParts(): Array[RuleInfo] =
+    ruleCache.getRule.collections.filter(_.catLevelTwo == Constants.default).toArray
+
+  override def runOnPart(builder: DiffGraphBuilder, collectionRuleInfo: RuleInfo): Unit = {
+    tagFuncCallCollection(builder, collectionRuleInfo)
+    tagRestCallCollection(builder, collectionRuleInfo)
+  }
+
+  private def tagFuncCallCollection(builder: DiffGraphBuilder, collectionRuleInfo: RuleInfo): Unit = {
+    val collectionCallMethod = cpg.call
+      .name("Methods")
+      .where(_.astChildren.isLiteral.code("(?i).*(Get|Post|Put|Patch|Delete).*"))
```

Comment:
create constant for `(?i).*(Get|Post|Put|Patch|Delete).*`

Reply:
done.